### DISCUSSION TITLE
[Feature] Add the ability to create API keys with limited permissions

### DISCUSSION THREAD

**Discussion by jwillikers** [immich-app/immich#6910] [Feature Request]
### The feature

Generated API keys have all permissions for an associated user. I'm working on a DIY camera, and I'd like to have it automatically upload photos via Immich CLI. The primary issue with this is that the API key has the scope of the user. This becomes really insecure when using my default admin user, but even using a regular user, the API key can probably still be used to delete uploaded assets. Ideally, I'd be able to generate an API key for my admin user account with upload-only permissions and be able to use this key on the camera.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mmomjian** [comment]
I would also love to have an API key restricted to monitoring, such as UptimeKuma. I currently use the `GET /server-info/statistics` endpoint to check for library sizes but some kind of health / status endpoint could be good as well.

**Comment by radh21301** [comment]
I definitely need this. I am using a couple of tools that work with immich. Immich-frame (digital photo frame) and Dawarich (google timeline alternative) - and I would really like to avoid scenarios where they could delete or corrupt my immich data since they have unrestricted access.

**Comment by bo0tzz** [comment]
This is now possible, it's just not exposed in the UI yet.

**Reply by bestrocker221** [reply]
Thats great! I can see in the API key documentation the possibility to specify the scope.
Is in the plan to add the support for even more granular permissions, for example read-only per-album?

**Reply by adamzvolanek** [reply]
Will it be exposed in the UI with the major release of Immich?

**Comment by wuzihao051119** [comment]
#18179
